libtransmission/rpcimpl.cc

```diff
@@ -148,7 +148,12 @@ auto getTorrents(tr_session* session, tr_variant* args)
     else // all of them
     {
         auto const& by_id = session->torrents().sorted_by_id();
-        torrents = std::vector<tr_torrent*>{ std::begin(by_id), std::end(by_id) };
+        torrents.reserve(std::size(by_id));
+        std::copy_if(
+            std::begin(by_id),
+            std::end(by_id),
+            std::back_inserter(torrents),
+            [](auto const* tor) { return tor != nullptr; });
```

Comment:
This change may not be necessary, at least not if any caller of the function correctly checks each entry of the vector for `NULL`.
But, we could skip the first entry here as well:
```suggestion
        torrents = std::vector<tr_torrent*>{ std::begin(by_id)+1, std::end(by_id) };
```

Reply:
There can be many `nullptr` values in between, not only the first one is null here.
Please read my comment https://github.com/transmission/transmission/pull/6177#issuecomment-1783956064

Reply:
I understand. But any method that access the returned list of `torrents` should check for `NULL` anyway, so an additional NULL check here might not be necessary.
Anyway, performance-wise it won't make much of a difference, I suppose?

Reply:
It won't matter performance wise.
But I would better fix it in one place according to the current invariant - no nulls returned - rather then update all 10+ call sites with null checks.

Reply:
@1100101 is right that the first element will always be nullptr and so it _could_ be skipped, but IMO hardcoding a `+1` here is RPC making minor optimizations based on knowledge of how `tr_torrents` is coded. I could imagine this biting us if `tr_torrents` ever gets refactored.